src/preamble.js

```diff
@@ -143,8 +143,9 @@ var cwrap, ccall;
       var ret = 0;
       if (str !== null && str !== undefined && str !== 0) { // null string
         // at most 4 bytes per UTF-8 code point, +1 for the trailing '\0'
-        ret = Runtime.stackAlloc((str.length << 2) + 1);
-        writeStringToMemory(str, ret);
+        var len = (str.length << 2) + 1;
```

Comment:
Or I suppose the 6-byte encoding scheme still stands, but Unicode guarantees that no valid Unicode code point will ever exceed a value that would require more than 4 bytes in UTF-8 to encode.